src/jbuild.ml

```diff
@@ -834,6 +837,22 @@ module Library = struct
        and project = Dune_project.get_exn ()
        and dune_version = Syntax.get_exn Stanza.syntax
        in
+       let name =
+         match name, public with
+         | Some n, _ -> n
+         | None, Some { name = (_loc, name) ; _ }  ->
+           if dune_version >= (1, 1) then
+             name
+           else
+             of_sexp_error loc "name field cannot be omitted before version 1.1"
```

Comment:
To be consistent with other error messages: `before version 1.1 of the dune language`